### PR TITLE
Implementation of BPMN message events

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2637

Audit message producer test coverage:

image

Delivery Message Rest Api test coverage:

image

### CODE REVIEW

**Comment by CTI777** [comment]
Tests are not done yet!

**Review by igdianov — APPROVED** [review]
LGTM

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/ProcessEngineEventsAggregator.java [hunk 1]
```diff
@@ -132,6 +133,8 @@ protected String resolveExecutionId(CloudRuntimeEvent<?, ?> element) {
             return ((CloudBPMNSignalEvent) element).getEntity().getProcessInstanceId();
         } else if(element instanceof CloudBPMNTimerEvent) {
             return ((CloudBPMNTimerEvent) element).getEntity().getProcessInstanceId();
+        } else if(element instanceof CloudBPMNMessageEvent) {
```
Comment (erdemedeiros) [inline activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/ProcessEngineEventsAggregator.java:136]: That's not the scope of this PR, but we need to review this in the future. This big `if/else` block is really ugly.

Comment (igdianov) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/ProcessEngineEventsAggregator.java:136]: @erdemedeiros I agree, We need to address this somehow so each concrete event provides its `processInstanceId` via `RuntimeEvent.getEntityProcessInstanceId` interface method. It should happen automatically once the event implementation provides method implementation via interface.

**Comment by igdianov** [comment]
@erdemedeiros  There was another problem with run-time error when sending start message to interrupting event subprocess, which we need to address in the near future.

**Pull request merged**